File: LeetCode/Easy/0141-linked-list-cycle/0141-linked-list-cycle.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
bool hasCycle(struct ListNode *head) {
    if(!head || !head->next)
        return false;
    struct ListNode *slow;
    struct ListNode *fast;
    slow = head;
    fast = head->next;
    while(slow != fast){
        if(!fast || !fast->next)
            return false;
        slow = slow->next;
        fast = fast->next->next;
    }
    return true;
}

File: LeetCode/Easy/0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        val_set = set()
        while head:
            if head in val_set:
                return True
            val_set.add(head)
            head = head.next
            # if head.next == None:
        return False